chapters/en/slides/module7/module7_04.Rmd

```diff
@@ -0,0 +1,342 @@
+
+Notes:
+Instead of creating a new programming language to store its code,
+Vega-Lite uses the established JSON file format.
+
+JSON is already widely used on the web as a storage format for text data,
+so this approach means that Vega-Lite does not need to reinvent the wheel.
+Instead, there are rules for which words are allows to appear in the JSON
+file for it to be valid Vega-Lite code,
+which is referred to as the "specification" (or "spec" for short).
+
+You can see an example of how this looks for an Altair visualization
+by using the `to_json` method as we have done in this slide.
+You can see that the words are the same we use in Altair,
+but the format of the JSON specification
+is a bit more verbose than  the Altair syntax.
+
+You can paste this JSON directly into Vega-Lite online editor
+and you will see that it renders correctly.
+We included a link for this in the slide that you can try.
+
+This is the same that happens if we click the "Open in Vega Editor" entry
+in the three dot action button that is automatically included in each Altair chart.
+
+If you look at the HTML file we saved in the previous step
+you will see that this JSON spec is included together with some additional code
+that tells your browser to use Vega-Lite's web libraries to render this JSON spec.
+
+---
+
+## Charts can be rendered offline
+
+
+- [Link to install the offline Vega-Lite extension in JupyterLab](https://altair-viz.github.io/user_guide/display_frontends.html#displaying-in-jupyterlab)
+- Code to include a fallback image representation of your chart for offline rendering:
+
+    ```
+    alt.renderers.enable('mimetype')
+    ```
+
+Notes:
+Since displaying the JSON spec correctly depends on a the Vega-Lite web libraries,
+this means that our charts will only render if we have an internet connection.
+
+If you are working offline and want to render charts in JupyterLab,
+there are a few different options.
+
+If you want interactive chart elements to work,
+you need to install a JupyterLab extension
+that downloads the the Vega-Lite rendering library to your machine.
+[The steps for installing this extension are outlined in the Altair documentation](https://altair-viz.github.io/user_guide/display_frontends.html#displaying-in-jupyterlab)
+
+If you are OK with seeing a static version of your charts when you are offline,
+you could instruct Altair to use the Vega-Lite libraries
+only if there is an active internet connection
+and otherwise use and image for the plot.
+To do this,
+you can include the code from this slide on top of your notebook,
+just after your import statements.
+
+---
+
+## What if we want to save our charts as images instead?
+
+- The three dots action buttong can be used to save images manually
+- `altair_saver` can be used to save charts programatically
+    - [Instructions for how to install and work with Altair saver can be found here](https://github.com/altair-viz/altair_saver/).
```

Comment:
Can you please add code so students know which functions/methods to use to do this?

Reply:
I added the section from the transcripts of that save is used as before just with a new fieloextension